### ISSUE TITLE
Sniff nodes seems to happen a lot ( too much)

### ISSUE THREAD

**Issue by jrots** [olivere/elastic#332]
### Which version of Elastic are you using?

[ ] elastic.v3 (for Elasticsearch 2.x)

### Please describe the expected behavior
Sniffnodes should only happen once every 15 minutes

### Please describe the actual behavior
Happens every second accross multiple goroutines

### Any steps to reproduce the behavior?

Hi I'm using an AppContext struct that I share across goroutines, 

I initially setup a connection to elastic once in the beginning of the program, 
func createElasticClient(context *AppContext) {
```
	es, err := elastic.NewClient(
		elastic.SetURL(context.Config.Elastic.Server...),
		elastic.SetMaxRetries(10),
		//		elastic.SetGzip(true),
	)
	if err != nil {
		if context.LocalMode {
			context.ElasticClient = es
		} else {
			// Handle error
			log.Fatal("elastic error:", err)
			panic("connect to elastic failed")
		}
	} else {
		context.ElasticClient = es
	}
}

```
afterwards I don't use NewClient anywhere, 
but I'm seeing a slowdown from calls to ES, 
when stracing I see that there are happening a lot of _nodes calls 

```
root@host:~# strace -p 35769 -f -s 1000 -tt -T  2>&1 | grep "_nodes"
[pid 35780] 15:43:59.212463 write(120, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.70:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000029>
[pid 35804] 15:43:59.213132 write(6, "GET /_nodes/http HTTP/1.1\r\nHost: elastic29:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000021>
[pid 35802] 15:43:59.213202 write(66, "GET /_nodes/http HTTP/1.1\r\nHost: elastic23:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35804] 15:43:59.213435 write(67, "GET /_nodes/http HTTP/1.1\r\nHost: elastic24:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35803] 15:43:59.213465 write(11, "GET /_nodes/http HTTP/1.1\r\nHost: elastic20:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35802] 15:43:59.213501 write(7, "GET /_nodes/http HTTP/1.1\r\nHost: elastic21:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35804] 15:43:59.213715 write(14, "GET /_nodes/http HTTP/1.1\r\nHost: elastic25:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35802] 15:43:59.213759 write(59, "GET /_nodes/http HTTP/1.1\r\nHost: elastic22:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35804] 15:43:59.213991 write(60, "GET /_nodes/http HTTP/1.1\r\nHost: elastic26:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35803] 15:43:59.214024 write(12, "GET /_nodes/http HTTP/1.1\r\nHost: elastic19:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35802] 15:43:59.214049 write(109, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.73:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35800] 15:43:59.214074 write(3, "GET /_nodes/http HTTP/1.1\r\nHost: elastic17:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35802] 15:43:59.214277 write(72, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.76:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35804] 15:43:59.214311 write(48, "GET /_nodes/http HTTP/1.1\r\nHost: elastic27:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35803] 15:43:59.214371 write(94, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.72:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35804] 15:43:59.214473 write(8, "GET /_nodes/http HTTP/1.1\r\nHost: elastic28:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 54359] 15:43:59.214491 write(115, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.71:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35803] 15:43:59.214550 write(96, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.74:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35802] 15:43:59.214581 write(81, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.75:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35800] 15:43:59.214625 write(77, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.79:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 54359] 15:43:59.214673 write(41, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.69:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35804] 15:43:59.214691 write(56, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.67:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 54359] 15:43:59.214827 write(104, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.78:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35802] 15:43:59.214965 write(62, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.77:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35783] 15:44:01.233085 write(94, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.72:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35803] 15:44:01.234931 write(6, "GET /_nodes/http HTTP/1.1\r\nHost: elastic29:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000022>
[pid 35803] 15:44:01.235152 write(12, "GET /_nodes/http HTTP/1.1\r\nHost: elastic19:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000022>
[pid 35803] 15:44:01.235369 write(11, "GET /_nodes/http HTTP/1.1\r\nHost: elastic20:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000021>
[pid 35803] 15:44:01.235587 write(7, "GET /_nodes/http HTTP/1.1\r\nHost: elastic21:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000021>
[pid 35803] 15:44:01.235801 write(59, "GET /_nodes/http HTTP/1.1\r\nHost: elastic22:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000022>
[pid 35803] 15:44:01.236017 write(66, "GET /_nodes/http HTTP/1.1\r\nHost: elastic23:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000021>
[pid 35803] 15:44:01.236231 write(67, "GET /_nodes/http HTTP/1.1\r\nHost: elastic24:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000019>
[pid 35803] 15:44:01.236443 write(14, "GET /_nodes/http HTTP/1.1\r\nHost: elastic25:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000021>
[pid 35803] 15:44:01.236658 write(60, "GET /_nodes/http HTTP/1.1\r\nHost: elastic26:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000020>
[pid 35803] 15:44:01.236870 write(48, "GET /_nodes/http HTTP/1.1\r\nHost: elastic27:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000021>
[pid 35803] 15:44:01.237081 write(8, "GET /_nodes/http HTTP/1.1\r\nHost: elastic28:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142) = 142 <0.000021>
[pid 35803] 15:44:01.237299 write(79, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.71:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000020>
[pid 35803] 15:44:01.237510 write(41, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.69:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000021>
[pid 35803] 15:44:01.237723 write(81, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.75:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000022>
[pid 35803] 15:44:01.237943 write(16, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.70:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000021>
[pid 35803] 15:44:01.238159 write(62, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.77:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000021>
[pid 35803] 15:44:01.238373 write(49, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.78:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000021>
[pid 35803] 15:44:01.238593 write(96, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.74:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000022>
[pid 35803] 15:44:01.238808 write(24, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.76:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000020>
[pid 35803] 15:44:01.239021 write(77, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.79:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000020>
[pid 35803] 15:44:01.239236 write(56, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.67:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000021>
[pid 35803] 15:44:01.239464 write(109, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.73:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146) = 146 <0.000022>
[pid 35783] 15:44:01.239642 write(3, "GET /_nodes/http HTTP/1.1\r\nHost: elastic17:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35782] 15:44:03.212404 write(120, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.70:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35802] 15:44:03.212578 write(6, "GET /_nodes/http HTTP/1.1\r\nHost: elastic29:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35782] 15:44:03.212607 write(81, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.75:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35779] 15:44:03.212664 write(66, "GET /_nodes/http HTTP/1.1\r\nHost: elastic23:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35800] 15:44:03.212686 write(62, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.77:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35802] 15:44:03.212743 write(67, "GET /_nodes/http HTTP/1.1\r\nHost: elastic24:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35782] 15:44:03.212814 write(77, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.79:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35800] 15:44:03.212880 write(109, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.73:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35779] 15:44:03.212909 write(3, "GET /_nodes/http HTTP/1.1\r\nHost: elastic17:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35777] 15:44:03.212931 write(79, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.71:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35782] 15:44:03.212953 write(49, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.78:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35802] 15:44:03.212974 write(14, "GET /_nodes/http HTTP/1.1\r\nHost: elastic25:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35802] 15:44:03.213134 write(60, "GET /_nodes/http HTTP/1.1\r\nHost: elastic26:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35782] 15:44:03.213157 write(121, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.72:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35779] 15:44:03.213210 write(12, "GET /_nodes/http HTTP/1.1\r\nHost: elastic19:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35802] 15:44:03.213440 write(48, "GET /_nodes/http HTTP/1.1\r\nHost: elastic27:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35800] 15:44:03.213472 write(56, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.67:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35782] 15:44:03.213517 write(7, "GET /_nodes/http HTTP/1.1\r\nHost: elastic21:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35802] 15:44:03.213625 write(8, "GET /_nodes/http HTTP/1.1\r\nHost: elastic28:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35800] 15:44:03.213725 write(11, "GET /_nodes/http HTTP/1.1\r\nHost: elastic20:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35780] 15:44:03.213821 write(136, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.69:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35779] 15:44:03.213881 write(59, "GET /_nodes/http HTTP/1.1\r\nHost: elastic22:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 142 <unfinished ...>
[pid 35777] 15:44:03.213911 write(24, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.76:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
[pid 35771] 15:44:03.213999 write(96, "GET /_nodes/http HTTP/1.1\r\nHost: 192.168.14.74:9200\r\nUser-Agent: elastic/3.0.43 (linux-amd64)\r\nAccept: application/json\r\nAccept-Encoding: gzip\r\n\r\n", 146 <unfinished ...>
^C
```

I probably can "fix" this by using SetSniff(false) 
but just want to share this that smth doesnt seem right (green state on the cluster btw)..

**Comment by olivere** [comment]
Yes, that seems odd. I'm looking into my own intrastructure to see what's going on...

**Comment by olivere** [comment]
It seems that [this line](https://github.com/olivere/elastic/blob/release-branch.v3/client.go#L683) is wrong. We should use the sniffer interval for the ticker, not the sniffer timeout. Can you check that in your application?

Change:

```go
// sniffer periodically runs sniff.
func (c *Client) sniffer() {
	c.mu.RLock()
	timeout := c.snifferInterval
	c.mu.RUnlock()

	ticker := time.NewTicker(timeout)
	defer ticker.Stop()

	for {
		select {
		case <-c.snifferStop:
			// we are asked to stop, so we signal back that we're stopping now
			c.snifferStop <- true
			return
		case <-ticker.C:
			c.sniff(timeout)
		}
	}
}
```

to

```go
// sniffer periodically runs sniff.
func (c *Client) sniffer() {
	c.mu.RLock()
	timeout := c.snifferTimeout
	interval := c.snifferInterval
	c.mu.RUnlock()

	ticker := time.NewTicker(interval)
	defer ticker.Stop()

	for {
		select {
		case <-c.snifferStop:
			// we are asked to stop, so we signal back that we're stopping now
			c.snifferStop <- true
			return
		case <-ticker.C:
			c.sniff(timeout)
		}
	}
}
```

**Comment by olivere** [comment]
Just released v2.0.51 and v3.0.44 with a fix. If this doesn't fix the issue, please respond here and I'll reopen.

Thanks for posting the issue!
